torch/csrc/jit/script/init.cpp

```diff
@@ -452,6 +460,23 @@ void initJitScriptBindings(PyObject* module) {
           }
           auto graph = tracer::createGraphByTracing(func, std::move(inputs), num_inputs);
           self.create_method(name, std::move(graph), std::move(parameters));
+      })
+      .def("graph_for", [](Module& self, py::args args) {
+        if (self.find_method("forward")) {
+          return self.get_method("forward").graph_for(createVariableTensorList(args));
+        }
+        throw std::runtime_error("Attempted to call graph_for on a Module without a compiled forward()");
+      })
+      .def("_forward", [](Module& self, py::args args) {
+        // We implement this in C++ to avoid incurring the pybind11 dispatch
+        // overhead twice: once to call into the method lookup for "forward"
+        // and once to actually invoke the method.
+        //
+        // There is a thin wrapper on top of this method in the C++ version of
+        // ScriptModule.
+        auto inputs = createVariableTensorList(args);
```

Comment:
It would be nice to have a single function that does these three lines for both forward and __call__ on Methods. We are going to update this site in the future to handle more data types, and having two site will add potential for divergence.